### PR TITLE
[native] Add retry for abort results

### PR DESCRIPTION
**Pull request by pranjalssh**
## Description
* If abortResults call fails, it leads to query failure. So we should retry them. We cannot simply fire and forget them as query may get stuck otherwise. Added tests for the same.

* Refactored tests to separate out case when producer always fails.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-native-execution/presto_cpp/main/PrestoExchangeSource.h [hunk 1]
```diff
@@ -212,15 +214,15 @@ class PrestoExchangeSource : public velox::exec::ExchangeSource {
   const std::shared_ptr<folly::IOThreadPoolExecutor> exchangeExecutor_;
 
   std::shared_ptr<http::HttpClient> httpClient_;
-  RetryState retryState_;
+  RetryState requestRetryState_;
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/PrestoExchangeSource.h:217]:
```
s/requestRetryState_/dataRetryState_/?
```

File: presto-native-execution/presto_cpp/main/PrestoExchangeSource.cpp [hunk 2]
```diff
@@ -380,44 +380,59 @@ void PrestoExchangeSource::acknowledgeResults(int64_t ackSequence) {
 }
 
 void PrestoExchangeSource::abortResults() {
+  bool abortIssued = false;
+  if (!abortResultsIssued_.compare_exchange_strong(abortIssued, true)) {
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/PrestoExchangeSource.cpp:384]:
```
if (abortResultsIssued_.exchange(true)) {
  return;
}
```

**Review by xiaoxmeng — APPROVED** [review]
@pranjalssh LGTM. Please shadow this in presto batch or deltoid a bit before land. Thanks!

File: presto-native-execution/presto_cpp/main/PrestoExchangeSource.cpp [hunk 3]
```diff
@@ -380,44 +380,59 @@ void PrestoExchangeSource::acknowledgeResults(int64_t ackSequence) {
...
           const std::string errMsg = fmt::format(
-              "Abort results failed: {}, path {}", statusCode, self->basePath_);
+              "Abort results failed: {}, path {}",
+              error.value(),
+              self->basePath_);
           LOG(ERROR) << errMsg;
-          onFinalFailure(errMsg, queue);
-        } else {
-          self->abortResultsSucceeded_.store(true);
+          return onFinalFailure(errMsg, queue);
```
[the remark this answers is not in the document]
Comment (pranjalssh) [reply presto-native-execution/presto_cpp/main/PrestoExchangeSource.cpp:428]: We need to call queue->setError() so we need it

**Review by mbasmanova — COMMENTED** [review]
@pranjalssh Would you update commit message and PR description to explain why it is necessary to retry abort API calls? Why can't we use fire-and-forget semantics for these?

**Pull request merged**